thrift/src/main/java/com/linecorp/armeria/server/thrift/ThriftCallService.java

```diff
@@ -166,13 +166,17 @@ private static void invokeSynchronously(
             }
 
             try {
-                final TBase<?, ?> result = f.getResult(impl, args);
                 if (func.isOneWay()) {
                     reply.complete(null);
+                    f.getResult(impl, args);
                 } else {
+                    final TBase<?, ?> result = f.getResult(impl, args);
                     reply.complete(func.getResult(result));
                 }
             } catch (Throwable t) {
+                if (func.isOneWay()) {
+                    throw new RuntimeException(t);
```

Comment:
The exception will go out of our hand if we rethrow here. How about logging instead?

```java
if (func.isOneWay()) {
    logger.warn("{} Unexpected exception from a one-way function:", ctx, t);
} else {
    reply.completeExceptionally(t);
}
```

Reply:
logging is more perfect. My first fix just want to log, But need to import logger, so, give up

Reply:
Thanks!

I believe `invokeAsynchronously()` should be changed like this:

```java
    private static final AsyncMethodCallback<Object> NOOP_CALLBACK = new AsyncMethodCallback<Object> {
        @Override
        public void onComplete(Object response) {}

        @Override
        public void onError(Exception e) {}
    };

    private static void invokeAsynchronously(
            Object impl, ThriftFunction func, TBase<?, ?> args, DefaultRpcResponse reply) throws TException {

        final AsyncProcessFunction<Object, TBase<?, ?>, Object> f = func.asyncFunc();
        if (func.isOneWay()) {
            f.start(impl, args, NOOP_CALLBACK);
            reply.complete(null);
        } else {
            f.start(impl, args, new AsyncMethodCallback<Object>() {
                @Override
                public void onComplete(Object response) {
                    reply.complete(response);
                }

                @Override
                public void onError(Exception e) {
                    reply.completeExceptionally(e);
                }
            });
        }
    }
```

Reply:
My opinion:
It is a bad way to use oneway call in async. Because async call donot wait the invoke result.Something like sync oneway call not interested with the result.And in Thrift, oneway Method Must be Void, the return has no scense.
But how to measure the oneway call complete?Just send out or up to the server side?
The sync oneway call change is the 'up to server side' way, for Unifying Concept， The invokeAsynchronously() change is the good way!!!
How to choose a more reasonable?@trustin

Reply:
Because it's a server-side change, the behavior has to be identical for sync and async calls, i.e. my proposed change in `invokeAsynchronously()` is required. Otherwise, a synchronous client that calls an asynchronous one-way function will block until the server finishes handling the request.

Reply:
yeah I see,  Agree! I update my fix with that 'proposed change' code!!!